api/analyzers/source_analyzer.py

```
@@ -146,24 +176,12 @@
-
         # Analyze source files
-        self.analyze_sources(ignore)
+        self.analyze_sources(Path(path), ignore, g)
 
         logging.info("Done analyzing path")
 
-        # Restore original working dir
-        os.chdir(original_dir)
-
-    def analyze_local_repository(self, path: str, ignore: Optional[List[str]] = []) -> Graph:
+    def analyze_local_repository(self, path: str, ignore: Optional[list[str]] = []) -> Graph:
```

Ruff (0.8.2)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)